beacon-chain/db/filesystem/iteration_test.go

```diff
@@ -0,0 +1,296 @@
+		children: []dirFiles{{name: "1.ssz"}, {name: "2.ssz"}},
+	}
+	blobWithSszAndTmp := dirFiles{name: rootStrs[2], isDir: true,
+		children: []dirFiles{{name: "5.ssz"}, {name: "0.part"}}}
+	tree.children = append(tree.children,
+		notABlob, childlessBlob, blobWithSsz, blobWithSszAndTmp)
+
+	topChildren := make([]string, len(tree.children))
+	for i := range tree.children {
+		topChildren[i] = tree.children[i].name
+	}
+
+	var filter = func(entries []string, filt func(string) bool) []string {
+		filtered := make([]string, 0, len(entries))
+		for i := range entries {
+			if filt(entries[i]) {
+				filtered = append(filtered, entries[i])
+			}
+		}
+		return filtered
+	}
+
+	tree.reify(t, fs, "")
+	cases := []struct {
+		name     string
+		dirPath  string
+		expected []string
+		filter   func(string) bool
+		err      error
+	}{
+		{
+			name:     "non-existent",
+			dirPath:  "derp",
+			expected: []string{},
+			err:      os.ErrNotExist,
+		},
+		{
+			name:     "empty",
+			dirPath:  childlessBlob.name,
+			expected: []string{},
+		},
+		{
+			name:     "top",
+			dirPath:  ".",
+			expected: topChildren,
+		},
+		{
+			name:     "custom filter: only notABlob",
+			dirPath:  ".",
+			expected: []string{notABlob.name},
+			filter: func(s string) bool {
+				return s == notABlob.name
+			},
+		},
+		{
+			name:     "root filter",
+			dirPath:  ".",
+			expected: []string{childlessBlob.name, blobWithSsz.name, blobWithSszAndTmp.name},
+			filter:   isRootDir,
+		},
+		{
+			name:     "ssz filter",
+			dirPath:  blobWithSsz.name,
+			expected: blobWithSsz.childNames(),
+			filter:   isSszFile,
+		},
+		{
+			name:     "ssz mixed filter",
+			dirPath:  blobWithSszAndTmp.name,
+			expected: []string{"5.ssz"},
+			filter:   isSszFile,
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			result, err := listDir(fs, c.dirPath)
+			if c.filter != nil {
+				result = filter(result, c.filter)
+			}
+			if c.err != nil {
+				require.ErrorIs(t, err, c.err)
+				require.Equal(t, 0, len(result))
+			} else {
+				require.NoError(t, err)
+				sort.Strings(c.expected)
+				sort.Strings(result)
+				require.DeepEqual(t, c.expected, result)
+			}
+		})
+	}
+}
+
+func TestSlotFromBlob(t *testing.T) {
+	cases := []struct {
+		slot primitives.Slot
+	}{
+		{slot: 0},
+		{slot: 2},
+		{slot: 1123581321},
+		{slot: math.MaxUint64},
+	}
+	for _, c := range cases {
+		t.Run(fmt.Sprintf("slot %d", c.slot), func(t *testing.T) {
+			_, sidecars := util.GenerateTestDenebBlockWithSidecar(t, [32]byte{}, c.slot, 1)
+			sc := sidecars[0]
+			enc, err := sc.MarshalSSZ()
+			require.NoError(t, err)
+			slot, err := slotFromBlob(bytes.NewReader(enc))
+			require.NoError(t, err)
+			require.Equal(t, c.slot, slot)
+		})
+	}
+}
+
+func TestIterationComplete(t *testing.T) {
+	targets := []migrationTestTarget{
+		{
+			ident: ezIdent(t, "0x0125e54c64c925018c9296965a5b622d9f5ab626c10917860dcfb6aa09a0a00b", 1234, 0),
```

Comment:
`ezIdent` is defined in `migration_test.go` but both used in `migration_test.go` and `iteration_test.go`.
As a common resource between 2 tests files, would'nt be useful to create a common test utils files? (Or to put this function in `mock.go`, even if it's not really a mock...)

Reply:
```suggestion
func TestIterationComplete(t *testing.T) {
	/*
by-epoch
├── 0
│   └── 1234
│       └── 0x0125e54c64c925018c9296965a5b622d9f5ab626c10917860dcfb6aa09a0a00b
│           └── 0.ssz
├── 1
│   └── 5330
│       └── 0x0127dba6fd30fdbb47e73e861d5c6e602b38ac3ddc945bb6a2fc4e10761e9a86
│           ├── 0.ss
│           └── 1.ssz
├── 2
│   └── 11235
│       └── 0x666cea5034e22bd3b849cb33914cad59afd88ee08e4d5bc0e997411c945fbc1d
│           └── 1.ssz
└── 4096
    ├── 16777216
    │   └── 0x0232521756a0b965eab2c2245d7ad85feaeaf5f427cd14d1a7531f9d555b415c
    │       ├── 0.ssz
    │       └── 1.ssz
    └── 16777217
        └── 0x42eabe3d2c125410cd226de6f2825fb7575ab896c3f52e43de1fa29e4c809aba
            └── 0.ssz
	*/

Reply:
(So it's easier to understand the test.)

Reply:
Generally I don't see a problem with this? If you imagine a rule where any type or func that is shared across 2 files needs to be defined in some special common file, that would require a lot of vigilance, decision fatigue and inconvenient separation. It's simple enough to go from a usage in one source to find it's definition in another. Don't feel super strongly about it but to me it feels like an uncomfortable convention to enforce.

Reply:
Yeah actually this is possible in golang, since all files in the same package are gathered into the same translation unit. In other languages, not having this kind of "tooling hygiene" could rapidly cause some circular import issues.

But that's fine to me!